app/channel.go

```diff
@@ -596,10 +596,40 @@ func (a *App) postChannelPrivacyMessage(user *model.User, channel *model.Channel
 	return nil
 }
 
-func (a *App) RestoreChannel(channel *model.Channel) (*model.Channel, *model.AppError) {
+func (a *App) RestoreChannel(channel *model.Channel, userId string) (*model.Channel, *model.AppError) {
```

Comment:
I agree with @hanzei, please add unit test for this.

Reply:
@saturninoabril  The functionality I am adding is validating and sending a NewWebSocketEvent.   I should add unit test to test this or the actual RestoreChannel functionality?

Reply:
Add unit test to RestoreChannel since to test yet on it.  Would be awesome if you could include check to websocket event.

Reply:
wrote a simple unit test.  i am not very experienced in writing unit test and not sure the best way to do it and what cases to test.  If i can get more guidance and help. that would be great.  Would this be better for a separate PR for the amount of work of writing good unit tests? I am also confused by the channel_test.go in api4 and app folder. there seems to be a lot of overlap